api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/Eri2MonthsCohortQueries.java

```diff
@@ -212,7 +212,7 @@ public CohortDefinition getChildrenRetainedOnArtFor2MonthsFromArtInitiation() {
         "children",
         EptsReportUtils.map(
             ageCohortQueries.createXtoYAgeCohort("Children 0 - 14", 0, 14),
-            "endDate=${cohortStartDate},location=${location}"));
+            "effectiveDate=${cohortStartDate}"));
```

Comment:
Should this be endDate instead of cohortStartDate?

Reply:
Good question. According to the spec it should be: 

> _a) Patients with birth date information registered in the system should be calculated the age of the patient at the date of initiation on ART  (birth date minus  ART initiation date ._ 
 Only with a calculation?

Reply:
Oh dear, then it is neither cohort or reporting endDate. We will need to use `InitialArtStartDateCalculation` to calculate the initaitedART date for the patient and use that instead.

Reply:
I think this will require a new cohort definition to use this calculation evaluator and I think it will require us to provide a location parameter for filtering the obs & encounters lookups.

Reply:
OK, will work on that

Reply:
https://github.com/esaude/openmrs-module-eptsreports/blob/eeda2b7a5407ddab25a759eac87e6f972f619375/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxNewCohortQueries.java#L300-L313 wont work?

Reply:
Please apply this change to both indicators, thanks

Reply:
@ynurmahomed that is based on the TxNew query, which funny enough is actually the same base query shared with these retention queries. So yes that could actually work...

If this report was not using the same base queries, then I don't think this would have worked. So in general we will need to use the initialDate calculation like was used for PVLS.

Reply:
@ynurmahomed that cohort is based on the txNew composition which is not exactly the same composition as for these retention indicators. So i don't think it will work. 

But it could have since these retention queries share the same base query as txNew

Reply:
OK. Doesn't it make sense that the ART Start date query be the same independent of indicator.

Reply:
I agree it should be, but the txNew cohorts are still using the old queries where the age calc was baked into the txNew query. Which is why Nicholas had to extract the logic into a separate calculator.

Moving forward we should always be using the same `InitialArtStartDateCalculation` calculation to determine ART initiation date in our cohort queries.